Document/0x05f-Testing-Authentication.md

```diff
@@ -4,47 +4,46 @@
 
 #### Overview
 
--- TODO [Provide a general description of the issue.] --
+Applications often have different sections with, on the one hand public and non-privileged information / actions, and on the other hand sensitive and privileged information / actions. Users can legitimately access the first ones without any restriction; however, in order to make sure sensitive and privileged information / actions are protected and accessible only to legitimate users, proper authentication has to take place.  
 
 #### Static Analysis
 
--- TODO [Describe how to assess this given either the source code or installer package (APK/IPA/etc.), but without running the app. Tailor this to the general situation (e.g., in some situations, having the decompiled classes is just as good as having the original source, in others it might make a bigger difference). If required, include a subsection about how to test with or without the original sources.] --
-
--- TODO [Confirm remark "Use the &lt;sup&gt; tag to reference external sources, e.g. Meyer's recipe for tomato soup<sup>[1]</sup>."] --
-
--- TODO [Develop content on Verifying that Users Are Properly Authenticated with source code] --
+When source code is available, first locate all sections with sensitive and privileged information / actions: they are the ones that need to be protected. Prior to accessing any item, the application must make sure the user is really who he pretends to and that he is allowed to access the section. Look for keywords in the targeted programming language that are used to authenticate a user or to retrieve and check an existing session token (for instance: KeyStore, SharedPreferences, ...).
 
 
 #### Dynamic Analysis
 
--- TODO [Describe how to test for this issue "Verifying that Users Are Properly Authenticated" by running and interacting with the app. This can include everything from simply monitoring network traffic or aspects of the app’s behavior to code injection, debugging, instrumentation, etc.] --
+The easiest way to check authentication on an App is to try to browse the app and access privileged sections. When this cannot be done manually, an automated crawler can be used (for instance, try to start an Activity that contains sensitive information with Drozer without providing authentication elements). 
+
+Sometimes, an intercepting proxy can be used to capture network traffic while being authenticated. Then, log out and try to replay requests while removing authentication information or not.
```

Comment:
As this is technically possible only in the case when the app is exchanging information with a backend server, I changed "Sometimes" for "In case the app is exchanging information with a backend server".